caffe2/CMakeLists.txt

```diff
@@ -163,6 +163,7 @@ endif()
 if (BUILD_PYTHON)
   # ---[ Python.
   add_library(caffe2_pybind11_state MODULE ${Caffe2_CPU_PYTHON_SRCS})
+  set_target_properties(caffe2_pybind11_state PROPERTIES COMPILE_FLAGS "-fvisibility=hidden")
```

Comment:
Hmm this one is a little bit concerning as Caffe2 explicitly relies on global visibility to get things properly registered. Do you know which thing is creating offending visibilities?

Reply:
It's taken from [here](https://github.com/pybind/pybind11/blob/master/docs/upgrade.rst#stricter-enforcement-of-hidden-symbol-visibility-for-pybind11-modules) which suggests setting that flag, and it resolves some compile-time warnings.

Note that this does only apply to the pybind wrapper libs, not anything else, so the general C2 need for visibility shouldn't be affected (and this runs all our tests without issue)